pkg/reconciler/taskrun/taskrun.go

```diff
@@ -988,7 +985,8 @@ func launchProvisioningTask(r *ReconcileTaskRun, ctx context.Context, tr *tekton
 	}
 
 	provision := tektonapi.TaskRun{}
-	provision.Name = kmeta.ChildName(tr.Name, "-provision")
+	// #nosec G401 -- MD5 used only for non-cryptographic uniqueness
+	provision.Name = kmeta.ChildName(tr.Name, fmt.Sprintf("-provision-%x", md5.Sum([]byte(address)))[:5])
```

Comment:
this could lead to a name longer than the maximum allowed (253 chars) and still break. A couple of suggestions  to cope with this edge-case:
* use a shorter suffix, like `-p`, instead of `-provision`
* Make sure the length of the generated name is shorter than 253. If not, truncate the name to `253-(length(hash(address))+1)` and add as suffix `-hash(address)`

Reply:
no, the `ChildName` is safe - it truncates the 1st part, keeping the suffix, until the length is fit into allowed 53 chars.